### PR TITLE
Add API function useAbility

### LINKED ISSUE #5: Add function for 'using' an ability
Our API needs to have a 'use ability' function for users to call. We need to implement the backend code for doing this.

bool useAbility(abilityId)

- Input ability ID
- Get ability from ID
- Get and pay cost from ability
- Return true if using the ability was 'successful' (i.e.: cost paid, etc). False otherwise.

For now, we don't need this connected to the network API, because it doesn't exist yet. Testing in C++ is sufficient.

### PR DESCRIPTION
**Pull request by nicole-angove**

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -34,3 +44,13 @@ cc_test(
...
+
+cc_test(
+  name = "apitest",
+  size = "small",
+  srcs = ["apitest.cpp"],
+  deps = [
+    "@com_google_googletest//:gtest_main",
+    ":api",
+  ],
+)
```
Comment (LukeAngove) [inline BUILD]: Missing newline at end of file, please add it.

File: api.h [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#ifndef API_H
+#define API_H
+
+bool useAbility(int abilityID);
```
Comment (LukeAngove) [inline api.h:9]: This is a public API function, so it should have documentation explaining what it is, what it does and what the parameters are. For example, start with something like:
```suggestion
/**
 * Tries to use the given ability.
 * 
 * Returns false if the ability cannot be used.
 */
bool useAbility(int abilityID);
```

Comment (LukeAngove) [reply api.h:9]: See https://google.github.io/styleguide/cppguide.html#Function_Comments for some good guidelines.

File: apitest.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#include <gtest/gtest.h>
+#include "api.h"
+
+TEST(API, useAbilitySuccess) {
+
```
Comment (LukeAngove) [inline apitest.cpp]: Remove this empty line.

**Review by LukeAngove — CHANGES_REQUESTED** [review]
Minor changes requested, just fixing white-space and adding some documentation.

**Review by LukeAngove — APPROVED** [review]

**Review by Johjahn1 — APPROVED** [review]

**Pull request merged**